reasoning/src/commonMain/kotlin/com/xebia/functional/xef/reasoning/serpapi/SerpApiClient.kt

```diff
@@ -0,0 +1,80 @@
+package com.xebia.functional.xef.reasoning.serpapi
+
+import com.xebia.functional.xef.env.getenv
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+
+class SerpApiClient : AutoCloseable {
```

Comment:
should implement `AutoClose by autoClose()` if we want it to close automatically with the `ai` block

Reply:
it has override close of autoClose() like in GcpClient.kt

```kotlin
override fun close() {
        http.close()
    }
``` 
it seems that implement is for abstract

![image](https://github.com/xebia-functional/xef/assets/14172789/eeafb6e6-428d-4cb2-84b3-3324d7f82be8)

Reply:
Done